drivers/display/ssd1327.c

```diff
@@ -0,0 +1,396 @@
+{
+	size_t buf_len;
+
+	if (desc->pitch < desc->width) {
+		LOG_ERR("Pitch is smaller then width");
+		return -1;
+	}
+	/* Following the datasheet, in the GDDRAM, two segment are split in one register */
+	buf_len = MIN(desc->buf_size, desc->height * desc->width / 2);
+	if (buf == NULL || buf_len == 0U) {
+		LOG_ERR("Display buffer is not available");
+		return -1;
+	}
+
+	if (desc->pitch > desc->width) {
+		LOG_ERR("Unsupported mode");
+		return -1;
+	}
+
+	if ((y & 0x7) != 0U) {
+		LOG_ERR("Unsupported origin");
+		return -1;
+	}
+
+	LOG_DBG("x %u, y %u, pitch %u, width %u, height %u, buf_len %u", x, y, desc->pitch,
+		desc->width, desc->height, buf_len);
+
+	return ssd1327_write_default(dev, x, y, desc, buf, buf_len);
+}
+
+static int ssd1327_set_contrast(const struct device *dev, const uint8_t contrast)
+{
+	uint8_t cmd_buf[] = {
+		SSD1327_SET_CONTRAST_CTRL,
+		contrast,
+	};
+
+	return ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+}
+
+static void ssd1327_get_capabilities(const struct device *dev,
+				     struct display_capabilities *caps)
+{
+	const struct ssd1327_config *config = dev->config;
+
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = config->width;
+	caps->y_resolution = config->height;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO10;
+	caps->current_pixel_format = PIXEL_FORMAT_MONO10;
+	caps->screen_info = SCREEN_INFO_MONO_VTILED;
+}
+
+static int ssd1327_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pf)
+{
+	if (pf == PIXEL_FORMAT_MONO10) {
+		return 0;
+	}
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+static int ssd1327_init_device(const struct device *dev)
+{
+	const struct ssd1327_config *config = dev->config;
+
+	uint8_t cmd_buf[] = {
+		SSD1327_SET_ENTIRE_DISPLAY_OFF,
+		(config->color_inversion ? SSD1327_SET_REVERSE_DISPLAY
+					 : SSD1327_SET_NORMAL_DISPLAY),
+	};
+
+	/* Reset if pin connected */
+	if (config->reset.port) {
+		k_sleep(K_USEC(SSD1327_RESET_DELAY_1000US));
+		gpio_pin_set_dt(&config->reset, 1);
+		k_sleep(K_USEC(SSD1327_RESET_DELAY_100US));
+		gpio_pin_set_dt(&config->reset, 0);
+		k_sleep(K_MSEC(SSD1327_RESET_DELAY_300MS));
+	}
+
+	/* Turn display off */
+	if (ssd1327_suspend(dev)) {
+		return -EIO;
+	}
+
+	if (ssd1327_set_display(dev)) {
+		return -EIO;
+	}
+
+	if (ssd1327_set_contrast(dev, CONFIG_SSD1327_DEFAULT_CONTRAST)) {
+		return -EIO;
+	}
+
+	if (ssd1327_set_hardware_config(dev)) {
+		return -EIO;
+	}
+
+	if (ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true)) {
+		return -EIO;
+	}
+
+	if (ssd1327_set_timing_setting(dev)) {
+		return -EIO;
+	}
+
+	ssd1327_resume(dev);
+
+	return 0;
+}
+
+static int ssd1327_init(const struct device *dev)
+{
+	const struct ssd1327_config *config = dev->config;
+
+	LOG_DBG("Initializing device %s", config->bus_name(dev));
+
+	if (!ssd1327_bus_ready(dev)) {
+		LOG_ERR("Bus device %s not ready!", config->bus_name(dev));
+		return -EINVAL;
+	}
+
+	if (config->reset.port) {
```

Comment:
You should check if the `reset` GPIO device is ready or not with `gpio_is_ready_dt(&config->reset)`